tavern/util/dict_util.py

```diff
@@ -301,12 +301,28 @@ def _format_err(which):
                 except KeyError:
                     logger.debug("Skipping comparing missing key %s due to strict=%s", key, strict)
         elif isinstance(expected_val, list):
-            if len(expected_val) != len(actual_val):
-                raise_from(exceptions.KeyMismatchError("Length of returned list was different than expected - expected {} items, got {} ({})".format(len(expected_val), len(actual_val), full_err())), e)
-
             # TODO
             # Check things in the wrong order?
 
+            if not strict:
+                # Iterate over list items to see if any of them match...
+                for i, e_val in enumerate(expected_val):
+                    for a_val in actual_val:
+                        try:
+                            check_keys_match_recursive(e_val, a_val, keys + [i], strict)
```

Comment:
Just a discussion point, but I can see someone raising an issue of wanting to strictly match a single object in a list! If we go down this route, are we going to end up slowly re-implementing pykwalify or JSON schema?

Reply:
I'm not sure what you mean by ' strictly match a single object in a list', isn't that what it's doing already?

Reply:
Although looking at it again it does need to be changed so that all expected items are present